akvo/iati/imports/__init__.py

```diff
@@ -0,0 +1,440 @@
+    return '{http://akvo.org/iati-activities}budget-item'
+    :param tag_or_attr: string
+    :return: the string prefixed with the akvo namespace
+    """
+    return "{{{}}}{}".format(settings.AKVO_NS, tag_or_attr)
+
+
+class ImportMapper(object):
+    """
+    The ImportMapper holds data that is used over the whole import of an activity, and provides a
+    number of helper methods for extracting and massaging the data from the XML.
+
+    The ImportMapper is not used directly but is sub-classed for each tag or group of tags that we
+    want to extract data from into one RSR model. The sub-class must implement the do_import()
+    method where the actual mapping between the IATI XML data and the RSR models takes place.
+
+    In the base class a bunch of helper methods are defined, for common use cases of extracting data
+    from the XML.
+    """
+    def __init__(self, iati_import_job, parent_elem, project, globals,
+                 related_obj=None):
+        """
+
+        :param iati_import_job: The IatiImportJob object for the current import
+        :param parent_elem: this is usually the iati-activity, but in some cases it is another
+            element that has a number of sub-elements of the same kind, e.g. transaction/sector
+        :param project: the project we are assembling the data for
+        :param globals: the attributes on the iati-activities tag
+        :param related_obj: Related fields on the RSR models are usually to the Project object, but
+            in some cases it is another object to which we need to make a relation. In those cases
+            we use related_obj
+        """
+        self.iati_import_job = iati_import_job
+        # most often parent_elem is the activity element
+        self.parent_elem = parent_elem
+        self.project = project
+        self.globals = globals
+        # used for relations that are not to Project
+        self.related_obj = related_obj
+        self.model = Project
+        self._log_objects = []
+
+    def get_or_create_organisation(self, ref='', name='', org_id=None, internal_id=None):
+        """
+        Looks for an organisation in the RSR database.
+        First the ref will be looked up in the Organisation.iati_org_id field. If this does not exist,
+        the name will be looked up in the Organisation.name and Organisation.long_name fields.
+        If none of these return a match, a new organisation will be created.
+
+        :param ref: String; the reference of the organisation that is specified in the IATI file.
+        :param name: String; the name of the organisation that is specified in the IATI file.
+        :return: Organisation instance or None
+        """
+        if not (ref or name or (org_id and internal_id)):
+            return None
+
+        if ref:
+            try:
+                return Organisation.objects.get(iati_org_id=ref)
+            except ObjectDoesNotExist:
```

Comment:
```except ObjectDoesNotExist:``` could also be written as ```except Organisation.DoesNotExist:``` to make it a bit more specific. Both work fine though.